### PR TITLE
Update libunwindstack fork to 215644709bd7bbda4c05a9db4d55e96be61bcf64

### PR DESCRIPTION
**Pull request by dpallotti**
This is the second latest commit on `libunwindstack`'s `master` branch.
The latest switches to `#pragma once`, which makes it large and not that
interesting, so I'm leaving it out of this change.

Bug: http://b/230436550

Test:
- Unit tests.
- Capture Trata and verify unwinding, also together with uprobes and user space
  instrumentation.
- Try mixed unwinding prototype on top of this (conflicts are minimal).

### CODE REVIEW

**Comment by dpallotti** [comment]
At the moment the first seven commits are https://github.com/google/orbit/pull/3621.

**Review by danielfenner — APPROVED** [review]
I merely looked at the changes in our code to be honest - these lgtm.

File: src/LinuxTracing/UprobesUnwindingVisitor.cpp [hunk 1]
```diff
@@ -37,17 +37,23 @@ static void SendFullAddressInfoToListener(TracerListener* listener,
 
   FullAddressInfo address_info;
   address_info.set_absolute_address(libunwindstack_frame.pc);
-  address_info.set_module_name(libunwindstack_frame.map_name);
+
+  // Careful: unwindstack::FrameData::map_info might contain nullptr.
```
Comment (florian-kuebler) [inline src/LinuxTracing/UprobesUnwindingVisitor.cpp:41]: I guess we also want to add tests for the case that map info is nullptr. Maybe in a different PR?

**Review by florian-kuebler — APPROVED** [review]
LGTM. Maybe adding some tests for the nullptr case would be nice. But can be also done in a separate PR.

File: src/LinuxTracing/UprobesUnwindingVisitor.cpp (hunk 1 above)
Comment (dpallotti) [reply src/LinuxTracing/UprobesUnwindingVisitor.cpp:41]: You're right, it didn't seem part of the merge. https://github.com/google/orbit/pull/3637 will add a test after this change.

**Pull request merged**